organizations/organizations.go

```diff
@@ -290,3 +290,78 @@ func UpdateLogo(w http.ResponseWriter, r *http.Request) {
 
 	utils.GetSuccess("organization logo updated successfully", nil, w)
 }
+
+
+// transfer workspace ownership
+func TransferOwnership(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	org_collection, member_collection := "organizations", "members"
+
+	org_Id := mux.Vars(r)["id"]
+
+	// Check if organization id is valid
+	orgId, err := primitive.ObjectIDFromHex(org_Id)
+	if err != nil {
+		utils.GetError(errors.New("invalid organization id"), http.StatusBadRequest, w)
+		return
+	}
+
+	// Check if organization id exists in the database
+	orgDoc, _ := utils.GetMongoDbDoc(org_collection, bson.M{"_id": orgId})
+	if orgDoc == nil {
+		utils.GetError(errors.New("organization does not exist"), http.StatusBadRequest, w)
+		return
+	}
+
+	requestData := make(map[string]string)
+	if err := utils.ParseJsonFromRequest(r, &requestData); err != nil {
+		utils.GetError(err, http.StatusUnprocessableEntity, w)
+		return
+	}
+
+	email := requestData["email"]
+	
+	// confirm if email supplied is valid
+	if !utils.IsValidEmail(strings.ToLower(email)) {
+		utils.GetError(errors.New("email is not valid"), http.StatusBadRequest, w)
+		return
+	}
+
+	orgMember, err := FetchMember(bson.M{"email": email})
+
+	if err != nil {
+		utils.GetError(errors.New("user not a member of any work pace"), http.StatusBadRequest, w)
+		return
+	}
+
+	// confirm if the proposed new owner is a member of the organization
+	memberDoc, _ := utils.GetMongoDbDoc(member_collection, bson.M{"org_id": org_Id, "_id": orgMember.ID})
```

Comment:
We should be able to do this with the email field since it is also unique in the collection instead of going the long route of fetching his detail then using the ID, funnily, the function you used could pull the first record it finds of the user with the supplied email. As you didn't add the org id tp the filter, the result can be from the wrong organization and it will mess with this query. Try using the email here to query. It should work. If it does, you get the correct data of the member you need. Then you do not need the function you created.

Reply:
bson.M{"org_id": org_Id, "_id": orgMember.ID})

It includes the org_id

Funny enough...during my debugging sessions, I used email to test this so ofcourse I know it works. I dont know the full logic behind this organization of a thing. So I considered the fact  the same email migjt be used in diff organizations. But I noticed the ID is somewhat unique.


I just considered now that the Getmember search should have included the org id as well



I'll switch this to email